x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/lucene/LuceneMaxOperatorTestCase.java

```diff
@@ -0,0 +1,199 @@
+            query = new MatchAllDocsQuery();
+        } else {
+            query = SortedNumericDocValuesField.newSlowRangeQuery(FIELD_NAME, Long.MIN_VALUE, Long.MAX_VALUE);
+        }
+        return new LuceneMaxFactory(List.of(ctx), c -> query, dataPartitioning, between(1, 8), FIELD_NAME, getNumberType(), limit);
+    }
+
+    public void testSimple() {
+        testSimple(this::driverContext);
+    }
+
+    public void testSimpleWithCranky() {
+        try {
+            testSimple(this::crankyDriverContext);
+            logger.info("cranky didn't break");
+        } catch (CircuitBreakingException e) {
+            logger.info("broken", e);
+            assertThat(e.getMessage(), equalTo(CrankyCircuitBreakerService.ERROR_MESSAGE));
+        }
+    }
+
+    private void testSimple(Supplier<DriverContext> contexts) {
+        int size = between(1_000, 20_000);
+        int limit = randomBoolean() ? between(10, size) : Integer.MAX_VALUE;
+        testMax(contexts, size, limit);
+    }
+
+    public void testEmpty() {
+        testEmpty(this::driverContext);
+    }
+
+    public void testEmptyWithCranky() {
+        try {
+            testEmpty(this::crankyDriverContext);
+            logger.info("cranky didn't break");
+        } catch (CircuitBreakingException e) {
+            logger.info("broken", e);
+            assertThat(e.getMessage(), equalTo(CrankyCircuitBreakerService.ERROR_MESSAGE));
+        }
+    }
+
+    private void testEmpty(Supplier<DriverContext> contexts) {
+        int limit = randomBoolean() ? between(10, 10000) : Integer.MAX_VALUE;
+        testMax(contexts, 0, limit);
+    }
+
+    private void testMax(Supplier<DriverContext> contexts, int size, int limit) {
+        DataPartitioning dataPartitioning = DataPartitioning.DOC; // randomFrom(DataPartitioning.values());
+        NumberTypeTest numberTypeTest = getNumberTypeTest();
+        LuceneMaxFactory factory = simple(numberTypeTest, dataPartitioning, size, limit);
+        List<Page> results = new CopyOnWriteArrayList<>();
+        List<Driver> drivers = new ArrayList<>();
+        int taskConcurrency = 2; // between(1, 8);
+        for (int i = 0; i < taskConcurrency; i++) {
+            DriverContext ctx = contexts.get();
+            drivers.add(new Driver(ctx, factory.get(ctx), List.of(), new TestResultPageSinkOperator(results::add), () -> {}));
+        }
+        OperatorTestCase.runDriver(drivers);
+        assertThat(results.size(), lessThanOrEqualTo(taskConcurrency));
+
+        for (Page page : results) {
+            assertThat(page.getPositionCount(), is(1));
+            assertThat(page.getBlockCount(), is(2));
+            final Block block = page.getBlock(0);
+            assertThat(block.getPositionCount(), is(1));
+            numberTypeTest.assertPage(block, page.getBlock(1));
+        }
+        // We can't verify the limit
+        if (size <= limit) {
+            numberTypeTest.assertMaxValue();
+        }
```

Comment:
We could pick up from here and make a `Driver` that has the `Max` in `FINAL` mode. It should be able to consume the result and spit out the real max. That feels like it'd be lovely proof this "looks up" with the rest of the infrastructure.

Reply:
Not sure if a follow you here :)

Reply:
Ah, sorry. I mean if, instead of capturing the result of the operator directly - if we instead put the Max or Min agg right after it that'd prove that we're making the right output format.

Reply:
Done, we are using an aggregator function to compute the result from the output pages.